apps/sms/views/shared/js/navigation.js

```diff
@@ -1,361 +1,921 @@
+    return Promise.reject(new Error(
+      'Couldn\'t find the container element for view ' + newView.name
+    ));
+  }
 
-    // sliding
-    promise = promise.then(
-      this.slide.bind(this, nextPanelInfo.wrapperPosition)
-    ).then(function changePanel() {
-      currentPanel = {
-        panel: panel,
-        args: args
-      };
-    });
+  var isGoingBack = oldView && oldView.previous === newView.name;
 
-    // afterLeave and afterEnter should not trigger errors
-    promise = promise.then(function() {
-      var promise = Promise.resolve();
+  newPanelElement.classList.remove('panel-prepare-activation');
 
-      if (currentPanelObject && currentPanelObject.afterLeave) {
-        promise = promise.then(
-          currentPanelObject.afterLeave.bind(currentPanelObject, transitionArgs)
-        ).catch(catchError);
-      }
+  var doSlideAnimation =
+    from && from.panel && newView &&
+    from.panel.behavior !== newView.behavior;
 
-      if (nextPanelObject.afterEnter) {
-        promise = promise.then(
-          nextPanelObject.afterEnter.bind(nextPanelObject, transitionArgs)
-        ).catch(catchError);
-      }
+  console.log('doSlideAnimation', doSlideAnimation);
+  if (doSlideAnimation) {
+    newPanelElement.classList.add('panel-will-activate', 'panel-active');
+    oldPanelElement.classList.add('panel-will-deactivate');
+
+    newPanelElement.classList.toggle('panel-animation-reverse', isGoingBack);
+    oldPanelElement.classList.toggle('panel-animation-reverse', isGoingBack);
 
-      return promise;
+    return waitForSlideAnimation(newPanelElement).then(() => {
+      oldPanelElement.classList.remove(
+        'panel-will-activate', 'panel-will-deactivate',
+        'panel-active', 'panel-animation-reverse'
+      );
+      newPanelElement.classList.remove(
+        'panel-will-deactivate', 'panel-will-activate',
+        'panel-animation-reverse'
+      );
+    }, () => {
+      oldPanelElement.classList.remove(
+        'panel-will-deactivate', 'panel-animation-reverse'
+      );
+      newPanelElement.classList.remove(
+        'panel-will-activate', 'panel-animation-reverse'
+      );
     });
+  }
 
-    promise = promise.then(
-      function resolved() {
-        this.transitionPromise = null;
-        if (nextPanelContainer === currentPanelContainer) {
-          return;
-        }
-        nextPanelContainer.setAttribute('aria-hidden', false);
-        if (currentPanelContainer) {
-          currentPanelContainer.setAttribute('aria-hidden', true);
-        }
-      }.bind(this),
-      function rejected(e) {
-        catchError(e);
-        this.transitionPromise = null;
-        return Promise.reject(new Error('Error while transitioning'));
-      }.bind(this)
-    );
+  if (oldView) {
+    oldPanelElement.classList.remove('panel-active');
+  }
 
-    promise.
-      then(() => this.emit('navigated', { panel: panel, args: args })).
-      then(nextQueuedPanel, nextQueuedPanel);
+  if (newView) {
```

Comment:
__nit:__ mmm, likely no need to check if ```newView``` is defined, if not there are several places above where exception will be thrown (the first is ```document.querySelector(`.panel-${newView.behavior}`);```)